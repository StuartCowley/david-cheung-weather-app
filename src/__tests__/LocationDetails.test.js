import React from "react";
import { render, screen } from "@testing-library/react";
import LocationDetails from "../components/LocationDetails";
import renderer from "react-test-renderer";

describe("LocationDetails", () => {
  test("Renders as expected", () => {
    const rendered = renderer.create(
      <LocationDetails city="testCity" country="testCountry" />
    );
    expect(rendered).toMatchSnapshot();
  });

  test("Assert city and location are present", () => {
    render(<LocationDetails city="testCity" country="testCountry" />);
    expect(screen.getByText("testCity, testCountry")).toBeTruthy(); //  if the text exists as expected with toBeTruthy()
    expect(screen.getByText("testCity, testCountry")).toBeInstanceOf(
      HTMLHeadingElement
    ); //  more specific with our tests
    expect(screen.getByText("testCity, testCountry")).toHaveClass(
      "location-details"
    );
  });
});
