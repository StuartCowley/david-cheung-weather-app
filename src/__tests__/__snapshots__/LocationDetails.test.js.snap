// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDetails Renders as expected 1`] = `
<h1
  className="location-details"
>
  testCity, testCountry
</h1>
`;
